lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -384,6 +384,259 @@ class DecomposeAtenReshapeOp : public OpRewritePattern<AtenReshapeOp> {
 };
 } // namespace
 
+namespace {
+// support static shape only
```

Comment:
This decomposition is quite complex. Can you add a string explaining how this decomposition works? Namely, how different cases are identified and what they are decomposed into